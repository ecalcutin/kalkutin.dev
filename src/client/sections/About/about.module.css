.about {
  display: flex;
  gap: 5rem;

  @media (max-width: 950px) {
    display: block;

    .pic {
      margin: 5rem auto 0;
    }
  }

  ul {
    display: grid;
    grid-template-columns: repeat(2, minmax(8rem, 28rem));
    grid-gap: 0 1.5rem;
    margin-top: 1.5rem;

    li {
      font-family: var(--font-mono);
      font-size: 1.3rem;

      position: relative;
      margin-bottom: 1rem;
      padding-left: 1.5rem;

      &:before {
        content: 'â–¹';
        color: var(--color-secondary);
        position: absolute;
        left: 0;
      }
    }
  }
}
