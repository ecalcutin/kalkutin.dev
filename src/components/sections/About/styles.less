@import "theme/mixins.less";

.about {
  max-width: 900px;

  div.inner {
    display: grid;
    grid-template-columns: 3fr 2fr;
    grid-gap: 50px;

    @media (max-width: 768px) {
      display: block;
    }
  }
}

.skillList {
  display: grid;
  grid-template-columns: repeat(2, minmax(140px, 200px));
  grid-gap: 0 10px;
  padding: 0;
  margin: 20px 0 0 0;
  overflow: hidden;
  list-style: none;

  li {
    position: relative;
    margin-bottom: 10px;
    padding-left: 20px;
    font-family: @font-mono;
    font-size: @font-size-xs;

    &:before {
      content: "â–¹";
      position: absolute;
      left: 0;
      color: @color-accent;
      font-size: @font-size-sm;
      line-height: 12px;
    }
  }
}

.heading {
  .numberedHeading();
}

.myPicture {
  position: relative;
  max-width: 300px;

  .wrapper {
    display: block;
    position: relative;
    width: 100%;
    border-radius: @border-radius;
    background-color: @color-accent;

    &:hover,
    &:focus {
      background: transparent;
      outline: 0;

      &:after {
        top: 15px;
        left: 15px;
      }

      .img {
        filter: none;
        mix-blend-mode: normal;
      }
    }

    .img {
      position: relative;
      border-radius: @border-radius;
      mix-blend-mode: multiply;
      filter: grayscale(100%) contrast(1);
      transition: @transition-all;
      width: 100%;
    }

    &:before,
    &:after {
      content: "";
      display: block;
      position: absolute;
      width: 100%;
      height: 100%;
      border-radius: @border-radius;
      transition: @transition-all;
    }

    &:before {
      top: 0;
      left: 0;
      background-color: @color-primary;
      mix-blend-mode: screen;
    }

    &:after {
      border: 2px solid @color-accent;
      top: 20px;
      left: 20px;
      z-index: -1;
    }
  }

  @media (max-width: 768px) {
    margin: 50px auto 0;
    width: 70%;
  }
}
