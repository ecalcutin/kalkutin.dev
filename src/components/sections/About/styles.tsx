import styled from 'styled-components';

export const StyledAboutSection = styled.section`
  max-width: 900px;

  .inner {
    display: grid;
    grid-template-columns: 3fr 2fr;
    grid-gap: 50px;

    @media (max-width: 768px) {
      display: block;
    }
  }
`;

export const StyledText = styled.div`
  ul.skills-list {
    display: grid;
    grid-template-columns: repeat(2, minmax(140px, 200px));
    grid-gap: 0 10px;
    padding: 0;
    margin: 20px 0 0 0;
    overflow: hidden;
    list-style: none;

    li {
      position: relative;
      margin-bottom: 10px;
      padding-left: 20px;
      font-family: var(--font-mono);
      font-size: var(--fz-xs);

      &:before {
        content: 'â–¹';
        position: absolute;
        left: 0;
        color: var(--green);
        font-size: var(--fz-sm);
        line-height: 12px;
      }
    }
  }
`;

export const StyledPic = styled.div`
  position: relative;
  max-width: 300px;

  @media (max-width: 768px) {
    margin: 50px auto 0;
    width: 70%;
  }

  .wrapper {
    ${({ theme }) => theme.mixins.boxShadow};
    display: block;
    position: relative;
    width: 100%;
    border-radius: var(--border-radius);
    background-color: var(--green);

    &:hover,
    &:focus {
      background: transparent;
      outline: 0;

      &:after {
        top: 15px;
        left: 15px;
      }

      .img {
        filter: none;
        mix-blend-mode: normal;
      }
    }

    .img {
      position: relative;
      border-radius: var(--border-radius);
      mix-blend-mode: multiply;
      filter: grayscale(100%) contrast(1);
      transition: var(--transition);
    }

    &:before,
    &:after {
      content: '';
      display: block;
      position: absolute;
      width: 100%;
      height: 100%;
      border-radius: var(--border-radius);
      transition: var(--transition);
    }

    &:before {
      top: 0;
      left: 0;
      background-color: var(--navy);
      mix-blend-mode: screen;
    }

    &:after {
      border: 2px solid var(--green);
      top: 20px;
      left: 20px;
      z-index: -1;
    }
  }
`;
