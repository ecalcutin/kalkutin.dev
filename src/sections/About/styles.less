.about {
  max-width: 900px;

  .inner {
    display: grid;
    grid-template-columns: 3fr 2fr;
    grid-gap: 50px;
    
    @media (max-width: 768px) {
      display: block;
    }
  }
}
