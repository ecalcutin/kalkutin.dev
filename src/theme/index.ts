export { default as theme } from './theme-styles';
export { default as GlobalStyle } from './global';
export { default as mixins } from './mixins';
