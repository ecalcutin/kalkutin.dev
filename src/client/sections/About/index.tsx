import React from 'react';

import { Section } from 'client/components/Section';

import profilePic from '../../assets/images/profile.webp';

import styles from './about.module.css';
import { Picture } from './components/Picture';

export const About: React.FC = () => {
  return (
    <Section title='About Me'>
      <div className={styles.about}>
        <div>
          <div>
            <p>
              Hello! My name is Evgenii and I enjoy creating things that live on
              the internet. My interest in web development started back in 2013
              when I decided to try making personal blog â€” turns out hacking
              together a custom reblog button taught me a lot about HTML &amp;
              CSS!
            </p>
            <p>
              Here are a few technologies I&apos;ve been working with recently:
            </p>
          </div>
          <ul>
            {skills.map((skill, index) => (
              <li key={index}>{skill}</li>
            ))}
          </ul>
        </div>
        <Picture
          className={styles.pic}
          src={profilePic}
          alt='Evgenii Kalkutin profile photo'
        />
      </div>
    </Section>
  );
};

const skills: Array<string> = [
  'Node.js',
  'JavaScript (ES6+)',
  'Typescript',
  'Nest.js',
  'MongoDB',
  'React',
  'Vue',
  'Angular',
  'RxJS',
  'Jest & Playwright',
  'Webpack',
  'Docker',
];
