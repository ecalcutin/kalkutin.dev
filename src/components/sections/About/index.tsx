import React from 'react';

import profilePic from 'assets/images/profile.jpg';

import { StyledAboutSection, StyledPic, StyledText } from './styles';

const About = () => {
  const skills = [
    'Node.js',
    'JavaScript (ES6+)',
    'Typescript',
    'Nest.js',
    'MongoDB',
    'React',
    'Vue',
    'Angular',
    'RxJS',
    'Jest & Cypress',
    'Webpack',
    'Docker',
  ];

  return (
    <StyledAboutSection id='about'>
      <h2 className='numbered-heading'>About Me</h2>

      <div className='inner'>
        <StyledText>
          <div>
            <p>
              Hello! My name is Evgenii and I enjoy creating things that live on
              the internet. My interest in web development started back in 2013
              when I decided to try making personal blog â€” turns out hacking
              together a custom reblog button taught me a lot about HTML &amp;
              CSS!
            </p>

            <p>
              Here are a few technologies I&apos;ve been working with recently:
            </p>
          </div>

          <ul className='skills-list'>
            {skills && skills.map((skill, i) => <li key={i}>{skill}</li>)}
          </ul>
        </StyledText>

        <StyledPic>
          <div className='wrapper'>
            <img className='img' src={profilePic} />
          </div>
        </StyledPic>
      </div>
    </StyledAboutSection>
  );
};

export default About;
