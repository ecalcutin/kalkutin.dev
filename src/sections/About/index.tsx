import React from "react";
import styles from './styles.less'

const About = () => {
  return <section className={styles.about}>
    
  </section>;
};

export default About;
