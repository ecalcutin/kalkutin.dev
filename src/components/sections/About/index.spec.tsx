import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { theme } from "../../../theme";

import About from "./index";

describe("About section", () => {
  it("should render correctly", () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <About />
      </ThemeProvider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
