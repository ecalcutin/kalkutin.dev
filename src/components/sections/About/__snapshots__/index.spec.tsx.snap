// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About section should render correctly 1`] = `
<section
  className="sc-bdnxRM FnUsK"
  id="about"
>
  <h2
    className="numbered-heading"
  >
    About Me
  </h2>
  <div
    className="inner"
  >
    <div
      className="sc-gtsrHT fLKucx"
    >
      <div>
        <p>
          Hello! My name is Evgheni and I enjoy creating things that live on the internet. My interest in web development started back in 2013 when I decided to try making personal blog â€” turns out hacking together a custom reblog button taught me a lot about HTML & CSS!
        </p>
        <p>
          Here are a few technologies I've been working with recently:
        </p>
      </div>
      <ul
        className="skills-list"
      >
        <li>
          Node.js
        </li>
        <li>
          JavaScript (ES6+)
        </li>
        <li>
          Typescript
        </li>
        <li>
          Nest.js
        </li>
        <li>
          React
        </li>
        <li>
          Vue
        </li>
        <li>
          Angular
        </li>
        <li>
          Cypress
        </li>
        <li>
          Jest
        </li>
        <li>
          Docker
        </li>
      </ul>
    </div>
    <div
      className="sc-dlnjwi evobQn"
    >
      <div
        className="wrapper"
      >
        <img
          className="img"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
